mlir/lib/Dialect/Rock/utility/transformMapUtils.cpp

```diff
@@ -974,3 +990,191 @@ Value mlir::rock::insertTransposeAndBroadcastTransforms(
   }
   return inp;
 }
+
+TransformMapAttr mlir::rock::invertTransformMap(
```

Comment:
I would rather make this a property of the TransformMapAttr to define its own inversion (optionally ofc -- where its possible to do so)
Any reason why we went on to create a utility function ?

Reply:
Inverses are only really needed for this one rewrite, so having them walking around all the time is kinda silly, I'd think.

Reply:
Ah just realizing TransformTypes are just enums. I was thinking they were classes which can hold the "property" that we can nicely attribute the inverse to.

Reply:
(So its structurally not possible do what I propose, but I'd be cautious about proliferation of utility functions where each type could independently define its inverse)